lib/modules/manager/npm/post-update/lerna.ts

```diff
@@ -115,18 +115,21 @@ export async function generateLockFiles(
       extraEnv.NPM_AUTH = env.NPM_AUTH;
       extraEnv.NPM_EMAIL = env.NPM_EMAIL;
     }
-    const lernaVersion =
+    const lernaConstraint =
       config.constraints?.lerna ??
-      getLernaVersion(lernaPackageFile, await lazyPgkJson.getValue());
+      getLernaConstraint(lernaPackageFile, await lazyPgkJson.getValue());
     if (
-      !is.string(lernaVersion) ||
-      (semver.valid(lernaVersion) && semver.gte(lernaVersion, '7.0.0'))
+      !is.string(lernaConstraint) ||
+      (semver.valid(lernaConstraint) && semver.gte(lernaConstraint, '7.0.0')) ||
+      (semver.validRange(lernaConstraint) &&
+        (semver.satisfies('7.0.0', lernaConstraint) ||
+          semver.satisfies('7.999.999', lernaConstraint)))
```

Comment:
This logic is added. A little hacky but should cover pretty much everyone. If I had more time I'd duplicate the containerbase util logic and resolve it fully